modules/tinymce/src/core/main/ts/api/AddOnManager.ts

```diff
@@ -12,71 +12,73 @@ import Editor from './Editor';
+ *     const renderUI = function() {
+ *         return {
+ *             editorContainer: div,
+ *             iframeContainer: iframe
+ *         };
+ *     };
+ *
+ *     // Return the renderUI function
+ *     return { renderUI };
+ * });
  */
 
 /**
- * Plugin base class, this is a pseudo class that describes how a plugin is to be created for TinyMCE. The methods below are all optional.
+ * TinyMCE plugin manager class. Subclass of AddOnManager. Allows for custom plugins to be added to TinyMCE.
  *
- * @class tinymce.Plugin
+ * @class tinymce.PluginManager
  * @example
- * tinymce.PluginManager.add('example', function(editor, url) {
- *     // Add a button that opens a window
- *     editor.addButton('example', {
- *         text: 'My button',
- *         icon: false,
- *         onclick: function() {
- *             // Open window
- *             editor.windowManager.open({
- *                 title: 'Example plugin',
- *                 body: [
- *                     {type: 'textbox', name: 'title', label: 'Title'}
- *                 ],
- *                 onsubmit: function(e) {
- *                     // Insert content when the window form is submitted
- *                     editor.insertContent('Title: ' + e.data.title);
- *                 }
- *             });
+ * tinymce.PluginManager.add('MyPlugin', function(editor, url) {
+ *     // Register a toolbar button that triggers an alert when clicked
+ *     // To show this button in the editor, include it in the toolbar setting
+ *     editor.ui.registry.addButton('myCustomToolbarButton', {
+ *         text: 'My Custom Button',
+ *         onAction: function () {
+ *             alert('Button clicked!');
  *         }
  *     });
  *
- *     // Adds a menu item to the tools menu
- *     editor.addMenuItem('example', {
- *         text: 'Example plugin',
- *         context: 'tools',
- *         onclick: function() {
- *             // Open window with a specific url
- *             editor.windowManager.open({
- *                 title: 'TinyMCE site',
- *                 url: 'http://www.tinymce.com',
- *                 width: 800,
- *                 height: 600,
- *                 buttons: [{
- *                     text: 'Close',
- *                     onclick: 'close'
- *                 }]
- *             });
+ *     // Register a menu item that triggers an alert when clicked
+ *     // To show this menu item in the editor, include it in the menu setting
+ *     editor.ui.registry.addMenuItem('myCustomMenuItem', {
+ *         text: 'My Custom Menu Item',
+ *         onAction: function() {
+ *             alert('Menu item clicked');
  *         }
  *     });
+ *
+ *     // Return plugin metadata
+ *     return {name: 'MyPlugin', url: 'https://mydocs.com/myplugin};
  * });
  */
 
+/**
+ * This class handles the loading of themes, plugins and other add-ons and their language packs.
+ * Superclass of ThemeManager and PluginManager.
```

Comment:
We probably need some other way to mention that `ThemeManager` and `PluginManager` are instances. Atm they are accessible vis:
```
tinymce.PluginManager
tinymce.ThemeManager
tinymce.AddOnManager.PluginManager
tinymce.AddOnManager.ThemeManager
```
The first 2 are aliases for the second 2. I don't have any good suggestions of how to document that atm :cry:

Reply:
I'll keep thinking on it too :(

Reply:
Are you happy with the relevant changes @lnewson ?